HdProduction.BuildService/Startup.cs

```diff
@@ -19,20 +22,41 @@ public Startup(IConfiguration configuration)
     {
       Configuration = configuration;
     }
-    
+
     // This method gets called by the runtime. Use this method to add services to the container.
     // For more information on how to configure your application, visit https://go.microsoft.com/fwlink/?LinkID=398940
     public void ConfigureServices(IServiceCollection services)
     {
       services.AddMemoryCache();
-      services.AddMessageQueue<ProjectRequiresSelfHostBuildMessageHandler>(Configuration.GetSection("MessageQueue"));
+ 
       services.AddSingleton<IDatabaseConnector>(new DatabaseConnector(Configuration.GetConnectionString("Db")));
       services.AddScoped<IHelpdeskBuildService, HelpdeskBuildService>(
         c => new HelpdeskBuildService(Configuration.GetValue<string>("Uris:HelpdeskHostSources")));
       services.AddScoped<IContentServiceClient, ContentServiceClient>(
         c => new ContentServiceClient(Configuration.GetValue<string>("Uris:ContentStorage")));
       services.AddScoped<ISourcesUpdateRepository, SourcesUpdateRepository>();
       services.AddScoped<IBuildsRepository, BuildsRepository>();
+
+      AddMessageQueue(services, Configuration.GetSection("MessageQueue"));
+    }
+    
+    private static void AddMessageQueue(IServiceCollection services, IConfigurationSection mqConfigurationSection)
+    {
+      if (mqConfigurationSection.GetValue<bool>("Enabled"))
+      {
+        services.AddSingleton<IRabbitMqConnection>(
```

Comment:
Are there places in build-service where we will use RebitMqConnection directly?

Reply:
no:)

Reply:
So, what this registration performs for?

Reply:
we need this registration, because Consumer and Publisher use this dependency